### ISSUE TITLE
Question: Remove Response Class From API Operation Docs

### ISSUE THREAD

**Issue by sherl0cks** [springfox/springfox#1633]
Version 2.5.0, as latest 2.6.1 had issues with mustache versions.

**tl;dr**: How can I achieve the behavior of the second screenshot without changing my method signature to return `ResponseEntity<Void>`?

I have a few operations that return 204 with the correlating empty response body by default. In error scenarios, they will return a body that has details on the error. The default generated docs will nonetheless produce an empty response class, as seen below:

![screenshot from 2017-01-13 16-47-05](https://cloud.githubusercontent.com/assets/1335137/21949517/e1a99fca-d9a7-11e6-932b-78248cee603e.png)

I've been able to remove this generated doc by changing my method signature to return `ResponseEntity<Void>`; however this is not desirable as I cannot return the error model in error scenarios. Example below:

![screenshot from 2017-01-13 16-47-18](https://cloud.githubusercontent.com/assets/1335137/21949542/138ba330-d9a8-11e6-80da-b8c068d09bd5.png)

**Comment by dilipkrish** [comment]
The prescribed way to model that in spring is to have an [`@ErrorHandler`](https://spring.io/blog/2013/11/01/exception-handling-in-spring-mvc). While inferring the error response is not automatically supported, you can annotate your using `@ApiResponses` with the types, for error.

**Comment by sherl0cks** [comment]
@dilipkrish thanks for prompt reply. I do have the API Responses annotated as below. It's that top "Response Class" section that I would like to get rid of since the expected return is 204.

``` java
   @ApiResponses(value = {
            @ApiResponse(code = 204, message = "`Engagement` saved with the given `id`", response = Void.class),
            @ApiResponse(code = 400, message = "Invalid `Engagement`", response = ErrorModel.class),
            @ApiResponse(code = 500, message = "Unexpected Error", response = ErrorModel.class)})
```

**Comment by sherl0cks** [comment]
@dilipkrish moving different Response Types to Exception Handlers did the trick. Thanks for the link! I'll start the repo for your prompt help. You can close this issue.
